dipy/reconst/shm.py

```diff
@@ -230,164 +232,205 @@ def real_sph_harm(m, n, theta, phi):
+        even order SH functions. Otherwise returns only even order SH
+        functions. Default: False
+    is_legacy: bool, optional
+        If true, uses MRtrix 0.2 SH basis definition, where the sqrt(2) factor
+        is omitted. Else, uses MRtrix 3 definition presented above.
 
     Returns
-    --------
-    y_mn : real float
-        The real harmonic $Y^m_n$ sampled at `theta` and `phi` as
-        implemented in mrtrix. Warning: the basis is Tournier et al.
-        2007 [2]_; 2004 [1]_ is slightly different.
+    -------
+    real_sh : real float
+        The real harmonic $Y^m_n$ sampled at `theta` and `phi`.
     m : array
-        The order of the harmonics.
-    n : array
         The degree of the harmonics.
+    n : array
+        The order of the harmonics.
 
     References
     ----------
-    .. [1] Tournier J.D., Calamante F., Gadian D.G. and Connelly A.
-           Direct estimation of the fibre orientation density function from
-           diffusion-weighted MRI data using spherical deconvolution.
-           NeuroImage. 2004;23:1176-1185.
-    .. [2] Tournier J.D., Calamante F. and Connelly A. Robust determination
+    .. [1] Tournier J.D., Calamante F. and Connelly A. Robust determination
            of the fibre orientation distribution in diffusion MRI:
            Non-negativity constrained super-resolved spherical deconvolution.
            NeuroImage. 2007;35(4):1459-1472.
-
+    .. [2] Tournier J-D, Smith R, Raffelt D, Tabbara R, Dhollander T,
+           Pietsch M, et al. MRtrix3: A fast, flexible and open software
+           framework for medical image processing and visualisation.
+           NeuroImage. 2019 Nov 15;202:116-137.
     """
-    m, n = sph_harm_ind_list(sh_order)
+    if is_full_basis:
+        m, n = sph_harm_full_ind_list(sh_order)
+    else:
+        m, n = sph_harm_ind_list(sh_order)
+
     phi = np.reshape(phi, [-1, 1])
     theta = np.reshape(theta, [-1, 1])
 
-    m = -m
-    real_sh = real_sph_harm(m, n, theta, phi)
-    real_sh /= np.where(m == 0, 1., np.sqrt(2))
+    # In the case where m < 0, Tournier bases
+    # consider |m| for the SH degree
+    sh = spherical_harmonics(np.abs(m), n, phi, theta)
+
+    real_sh = np.where(m < 0, sh.imag, sh.real)
+    if not is_legacy:
+        real_sh *= np.where(m == 0, 1., np.sqrt(2))
+    else:
+        deprecate_with_version('The legacy tournier07 basis is outdated and \
+                                will be deprecated in a future release of \
+                                DIPY. Consider using the new tournier07 \
+                                basis.')
+
     return real_sh, m, n
 
 
-def real_full_sh_mrtrix(sh_order, theta, phi):
-    """
-    Compute real spherical harmonics as in Tournier 2007 [2]_, where the real
-    harmonic $Y^m_n$ is defined to be::
+def real_sh_descoteaux(sh_order, theta, phi,
+                       is_full_basis=False,
+                       is_legacy=False):
```

Comment:
```suggestion
                       legacy=False):
```

Reply:
Small thing, but I don't think that you need the `is_` in these variable names.

Reply:
I agree, I'll remove them and update the documentation.